### DISCUSSION TITLE
Migrations Warning

### DISCUSSION THREAD

**Discussion by casey-chow** [directus/directus#5884] [Feature Requests]
Hey there! Something I noticed while administering a copy of Directus is that migrations do pose a non-zero risk when deploying a new version, and it's sometimes difficult to tell if deploying a given version would lead to a migration. What would your thoughts be on introducing either a guide to tell when a version change has a change to the underlying schema or (more direct, but definitely more maintenance work) introduce a warning whenever a change to the schema is introduced?

**Comment by rijkvanzanten** [comment]
> and it's sometimes difficult to tell if deploying a given version would lead to a migration

Migrations aren't automatically run on update or startup 🤔 I do agree that it would be nice if the API would show a "Missing migrations" warning on startup though

**Reply by casey-chow** [reply]
> Migrations aren't automatically run on update or startup 🤔

I did the common Heroku thing of migrating on release 😅

**Reply by rijkvanzanten** [reply]
I'm doing an experiment with [changesets](https://github.com/atlassian/changesets) right now. By having a more in-depth changelog on the release notes and by me remembering to add some ⚠️'s around for migrations, I think we can alleviate most of this confusion 🤔

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
